src/vs/platform/userDataSync/common/userDataSync.ts

```diff
@@ -533,6 +533,7 @@ export interface IUserDataSyncEnablementService {
 
 	readonly onDidChangeResourceEnablement: Event<[SyncResource, boolean]>;
 	isResourceEnabled(resource: SyncResource): boolean;
+	getResourceEnablement(resource: SyncResource): boolean | undefined;
```

Comment:
What is the need of the new method `getResourceEnablement` here? Can this be implemented in existing `isResourceEnabled` method?

Reply:
Unfortunately no, we have to know if the value was explicitly changed by a user (e.g., `undefined` means never updated), but the existing method does not expose that information 🤷 

One idea I had is to change the existing method to `isResourceEnabled(resource: SyncResource, fallback: boolean = true): boolean;` so we could pass the fallback value dynamically. That is not ideal either because the caller would always have to know what resource has what default value and can become messy at some point 🤔

Reply:
I would prefer this approach rather `isResourceEnabled(resource: SyncResource, defaultValue?: boolean): boolean;` and call this method with fallback at the following place

https://github.com/microsoft/vscode/pull/242360/files#diff-bb39d7c36591ac930515684707ed163977cb7cd81bef278f2633b471397c3ad0R544

```
quickPick.selectedItems = items.filter(item => this.userDataSyncEnablementService.isResourceEnabled(item.id, true));
```

In the implementation of `isResourceEnabled` the default value will be false for Prompts

Reply:
@sandy081 ported over your recommendations, let me know what you think.